### PR TITLE
Add rubocop, simplecov, Travis. Auto-fix a ton of things... Take 2

### PR DESCRIPTION
**Pull request by highb**
After the feedback and review in #37 , I'm just going to make a new PR so we don't lose the discussion history in that PR (since I did some git history re-writing).

linteriscoming

### CODE REVIEW

File: lib/beaker-answers/answers.rb [hunk 1]
```diff
@@ -64,19 +63,19 @@ class Answers
         'orchestrator_database_user'     => DEFAULT_ANSWERS[:q_orchestrator_database_user],
         'orchestrator_database_password' => DEFAULT_ANSWERS[:q_orchestrator_database_password],
         'use_application_services'       => true,
-      }
+      },
```
Comment (tvpartytonight) [inline lib/beaker-answers/answers.rb]: Hmm, is this a style that we want to go with, the trailing comma for the last key-pair of a hash?

It's not my favorite thing, but, I'm not really against it either.

Comment (highb) [reply lib/beaker-answers/answers.rb]: It makes it a lot easier to add new key-pairs. Not strongly in favor, but it does seem convenient.

**Comment by highb** [comment]
Hm, that spec failure is new. Must be something introduced from the last time I auto-corrected. 👀

File: lib/beaker-answers/versions/version20162.rb [hunk 2]
```diff
@@ -103,7 +103,7 @@ def answer_hiera
       hocon.render
     end
 
-    def installer_configuration_string(host)
+    def installer_configuration_string(_host)
```
Comment (tvpartytonight) [inline lib/beaker-answers/versions/version20162.rb]: Why is this changed to `_host`? And, it doesn't seem to be referenced in the `answer_hiera` method that this method wraps, so...maybe it can just be removed? The argument, I mean...

**Comment by tvpartytonight** [comment]
There's quite a few errors @highb , 135 to be exact...so, I'll let you figure out what the issue might be before reviewing further, and I'm happy to pair if you want a second set of eyes. A few things of note:

- The errors for the versions of PE28, 3.0, 3.2...maybe we should just drop some of those tests? They don't seem to provide any value at this point.
- The errors seems to be originating from the 2016.2 class and propagating out, so fixing the issues there will probably cascade out and fix issues on the subclassed versions.

File: lib/beaker-answers/versions/version20162.rb (hunk 2 above)
Comment (highb) [reply lib/beaker-answers/versions/version20162.rb]: Yeah, that's rubocop's idea; if a variable is unused, it marks with with a `_` so that you can review it and decide if it is even necessary.

**Comment by highb** [comment]
Yeah, I'll be taking another look at these spec errors when I have a chance.

**Comment by kevpl** [comment]
@highb were you able to find time to get around to this?

**Comment by kevpl** [comment]
also, would it be easier to deal with this or get back to it if it was split into a few PRs? It would definitely be nicer to review... 👀

**Comment by highb** [comment]
That's a good point. Maybe the way forward on this is to add all of the current issues to the rubocop_todo.yml, then make further PRs removing exceptions from there and auto-correcting them.

**Comment by highb** [comment]
@kevpl Alright, I've removed the autocorrects and disabled the linter in Travis so that we can work incrementally on the lint fixes that we want in separate PRs, rather than all at once.

**Comment by kevpl** [comment]
@highb I assume that since this is for new code being submitted that a new release isn't needed?

**Comment by highb** [comment]
@kevpl Yeah, I haven't changed any code yet. That'll come in future PRs. :)
